src/cobra/util/context.py

```diff
@@ -1,46 +1,68 @@
 """Context manager for the package."""
 
 from functools import partial
+from typing import Any, Callable
 
 
 class HistoryManager:
-    """Record a list of actions to be taken at a later time. Used to
-    implement context managers that allow temporary changes to a
-    :class:`~cobra.core.Model`.
+    """
+    Define a base context manager.
+
+    It records a list of actions to be taken at a later time.
+    This is used to implement context managers that allow temporary
+    changes to a `cobra.core.Model`.
 
     """
 
     def __init__(self):
-
-        # self._history just acts as a stack
+        """Initialize the class."""
+        # this acts like a stack
         self._history = []
 
-    def __call__(self, operation):
-        """Add the corresponding method to the history stack.
+    def __call__(self, operation: Callable[[Any], Any]):
+        """Add the corresponding operation to the history stack.
 
         Parameters
         ----------
-        operation : `function`
-            A function to be called at a later time
+        operation : callable
+            A function to be called at a later time.
 
         """
-
         self._history.append(operation)
 
     def reset(self):
-        """Trigger executions for all items in the stack in reverse order"""
+        """Trigger executions for all items in the stack in reverse order."""
         while self._history:
             entry = self._history.pop()
             entry()
 
 
 def get_context(obj):
```

Comment:
I'm not sure about the context type right now.

```suggestion
def get_context(obj) -> Optional[Any]:
```

Reply:
Should be `HistoryManager`.